cmake/CPM_boost.cmake

```diff
@@ -0,0 +1,35 @@
+
+set(ENV{CPM_SOURCE_CACHE} "${CMAKE_SOURCE_DIR}/tt_metal/third_party/boost")
+
+include(${CMAKE_SOURCE_DIR}/cmake/CPM.cmake)
+set(BoostPackages
+    Align
```

Comment:
we don't use all of these packages directly, CPM can't figure out the transitive dependencies for boost itself?

Reply:
As far as I can tell, CPM is not able to figure out the transitive dependencies.